src/shacl/ShaclParser.ts

```diff
@@ -0,0 +1,583 @@
+    this.CONSUME(turtleTokenMap.LBracket);
+    this.CONSUME(this.shaclTokenMap.SHACL_alternativePath);
+    this.SUBRULE(this.shaclPropertyPathPath); // This does not match the SHACL spec, but it does match the test cases, which violate the spec. ;_;
+    this.OPTION(() => this.CONSUME(turtleTokenMap.Semicolon));
+    this.CONSUME(turtleTokenMap.RBracket);
+  });
+
+  shaclInversePath = this.RULE('shaclInversePath', () => {
+    this.CONSUME(turtleTokenMap.LBracket);
+    this.CONSUME(this.shaclTokenMap.SHACL_inversePath);
+    this.SUBRULE(this.shaclPropertyPathPath);
+    this.OPTION(() => this.CONSUME(turtleTokenMap.Semicolon));
+    this.CONSUME(turtleTokenMap.RBracket);
+  });
+
+  shaclZeroOrMorePath = this.RULE('shaclZeroOrMorePath', () => {
+    this.CONSUME(turtleTokenMap.LBracket);
+    this.CONSUME(this.shaclTokenMap.SHACL_zeroOrMorePath);
+    this.SUBRULE(this.shaclPropertyPathPath);
+    this.OPTION(() => this.CONSUME(turtleTokenMap.Semicolon));
+    this.CONSUME(turtleTokenMap.RBracket);
+  });
+
+  shaclOneOrMorePath = this.RULE('shaclOneOrMorePath', () => {
+    this.CONSUME(turtleTokenMap.LBracket);
+    this.CONSUME(this.shaclTokenMap.SHACL_oneOrMorePath);
+    this.SUBRULE(this.shaclPropertyPathPath);
+    this.OPTION(() => this.CONSUME(turtleTokenMap.Semicolon));
+    this.CONSUME(turtleTokenMap.RBracket);
+  });
+
+  shaclZeroOrOnePath = this.RULE('shaclZeroOrOnePath', () => {
+    this.CONSUME(turtleTokenMap.LBracket);
+    this.CONSUME(this.shaclTokenMap.SHACL_zeroOrOnePath);
+    this.SUBRULE(this.shaclPropertyPathPath);
+    this.OPTION(() => this.CONSUME(turtleTokenMap.Semicolon));
+    this.CONSUME(turtleTokenMap.RBracket);
+  });
+
+  shaclLiteralConstraint = this.RULE('shaclLiteralConstraint', () => {
+    this.OR([
+      {
+        ALT: () => this.SUBRULE(this.shaclIntConstraint),
+      },
+      // TODO: Some specificity here is possibly unnecessary.
+      {
+        ALT: () => this.SUBRULE(this.shaclStringConstraint),
+      },
+      {
+        ALT: () => this.SUBRULE(this.shaclStringLiteralQuoteConstraint),
+      },
+      {
+        ALT: () => this.SUBRULE(this.shaclLangStringConstraint),
+      },
+      {
+        ALT: () => this.SUBRULE(this.shaclBooleanConstraint),
+      },
+      {
+        ALT: () => this.SUBRULE(this.shaclAnyLiteralConstraint),
+      },
+    ]);
+  });
+
+  shaclIntConstraint = this.RULE('shaclIntConstraint', () => {
+    this.CONSUME(categoryTokenMap.IntTakingPredicate);
+    this.OR([
+      {
+        ALT: () => this.CONSUME(turtleTokenMap.INTEGER),
+      },
+      {
+        ALT: () => this.SUBRULE(this.shaclXsdInteger),
+      },
+    ]);
+  });
+
+  shaclStringConstraint = this.RULE('shaclStringConstraint', () => {
+    this.OR([
+      {
+        ALT: () => this.CONSUME(this.shaclTokenMap.SHACL_select),
+      },
+      {
+        ALT: () => this.CONSUME(this.shaclTokenMap.SHACL_ask),
+      },
+    ]);
+    this.SUBRULE(this.String); // TODO: a bit too lax?
```

Comment:
could/should we parse the actual query in here?

Reply:
Could we? Yes. Should we? I dunno. But I was thinking it's not required for MVP. 🤷‍♂